arch/arm/core/cortex_m/Kconfig

```diff
@@ -50,6 +50,7 @@ config CPU_CORTEX_M23
 	select CPU_CORTEX_M
 	# Omit prompt to signify "hidden" option
 	select ARMV8_M_BASELINE
+	select ARMV8_M_SE if CPU_HAS_SAU
```

Comment:
Feels a bit backwards at a glance to select the CPU type based on available hardware features instead of the other way around.

Reply:
Well, ARMV8_M_SE is not, exactly a CPU type; it rather signifies that the ARMv8-M CPU has the security extension. And this true (and only true) if the ARMv8 CPU has a security attribution unit.

We do similar stuff for CPU_HAS_FPU and CPU_HAS_MPU, at least for ARM :)

Reply:
Alright, makes sense.

Was thinking you could turn it into a straight sequence of `select`s with something like this, but not sure it'd actually be an improvement.

```
config CPU_CORTEX_M23_SE
	bool
	select CPU_CORTEX_M
	select ARMV8_M_BASELINE
	select ARMV8_M_SE

config ARMV8_M_SE
	bool
	...
	select CPU_HAS_SAU
```

Not important in this PR at least.

Reply:
Maybe some selects could be moved to the `CPU_CORTEX_M` symbol too.

Reply:
@ulfalizer . thanks for your comments in the review.
>config CPU_CORTEX_M23_SE
	bool
	select CPU_CORTEX_M
	select ARMV8_M_BASELINE
	select ARMV8_M_SE

Such a scheme would be great if we had a standard inheritance hierarchy. However, the security extension (_SE) is not the only extension for ARmv8. E.g. for M33, i.e we would also need to have CORTEX_M33_FP, CORTEX_M33_DSP and combinations of those.

Having the extensions as "selected" symbols increases modularity, making it easy also inside the source code.

Finally, as far as I can see, CPU_CORTEX_M really contains all commit symbols.